spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerEndpoint.java

```diff
@@ -340,6 +340,11 @@ public void setAckDiscarded(boolean ackDiscarded) {
 		this.ackDiscarded = ackDiscarded;
 	}
 
+	public void useRecordFilterStrategy(RecordFilterStrategy<? super K, ? super V> recordFilterStrategy) {
+		setRecordFilterStrategy(recordFilterStrategy);
+		setAckDiscarded(true);
```

Comment:
This new method does not make too much sense since we simply can do this `setAckDiscarded(true)` in the existing `setRecordFilterStrategy()`.
WDYT?

Reply:
I agree with you, i made new commit to follow your guideline. 
It is good point! 👍 

I was just worried that including `setAckDiscarded(true)` in `setFilterStrategy(...)` might bring unintended side effects. For example, a developer might not expect that `ackDiscarded` would change when updating `setFilterStrategy()`, since it's not specified in the method signature.